internal/api/keppel/quotas_test.go

```diff
@@ -213,7 +214,7 @@ func TestQuotasAPI(t *testing.T) {
 			RepositoryID:        1,
 			Digest:              test.DeterministicDummyDigest(idx),
 			VulnerabilityStatus: models.PendingVulnerabilityStatus,
-			NextCheckAt:         time.Unix(0, 0),
+			NextCheckAt:         None[time.Time](),
```

Comment:
Test objects should be broadly representative of how we expect the model to be used in practice. Since this test record has vulnerability status `Pending`, it should have a `NextCheckAt != nil`.
```suggestion
			NextCheckAt:         Some(time.Unix(0, 0)),
```

Same for the other test records that get changed in this PR.

In fact, this is a good candidate for a database-level constraint. Please add [a check constraint](https://www.postgresql.org/docs/current/ddl-constraints.html) that allows NextCheckAt to be nil if and only if the vulnerability status is Rotten. If you're unfamiliar with the respective PostgreSQL DDL, Daniel can help you.